src/crypto/CHIPCryptoPALmbedTLS.cpp

```diff
@@ -1405,39 +1405,62 @@ CHIP_ERROR ExtractAKIDFromX509Cert(const ByteSpan & certificate, MutableByteSpan
     return ExtractKIDFromX509Cert(false, certificate, akid);
 }
 
-CHIP_ERROR ExtractVIDFromX509Cert(const ByteSpan & certificate, VendorId & vid)
+namespace {
+
+CHIP_ERROR ExtractDNAttributeFromX509Cert(bool isVID, const ByteSpan & certificate, uint16_t & id)
 {
 #if defined(MBEDTLS_X509_CRT_PARSE_C)
     CHIP_ERROR error = CHIP_NO_ERROR;
     mbedtls_x509_crt mbed_cert;
+    mbedtls_asn1_named_data * asn1IdEntry;
 
     mbedtls_x509_crt_init(&mbed_cert);
 
     int result = mbedtls_x509_crt_parse(&mbed_cert, Uint8::to_const_uchar(certificate.data()), certificate.size());
     VerifyOrExit(result == 0, error = CHIP_ERROR_INTERNAL);
 
     // vendor id is the second element of subject and should be of size 4
+    // product id is the third element of subject and should be of size 4
```

Comment:
This method is dangerous, since it assumes the OID is correct in the correct order. The OIDs must be *explicitly checked*

Reply:
Agree with @tcarmelveilleux. We shouldn't assume specific order of attributes. I suggest defining:
`
constexpr uint8_t sOID_AttributeType_ChipVendorId[] = { 0x2B, 0x06, 0x01, 0x04, 0x01, 0x82, 0xA2, 0x7C, 0x02, 0x01 };
constexpr uint8_t sOID_AttributeType_ChipProductId[] = { 0x2B, 0x06, 0x01, 0x04, 0x01, 0x82, 0xA2, 0x7C, 0x02, 0x02 };
`
and do explicit check with each attribute. Similar to how it is done in `ExtractKIDFromX509Cert()`